modules/microshift-lvms-csi-snapshot-optional.adoc

```diff
@@ -0,0 +1,103 @@
+// Module included in the following assemblies:
+//
+// * microshift_storage/microshift-storage-plugin-overview.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-lvms-csi-snapshot-optional_{context}"]
+= Making LVMS and CSI volume snapshot deployment optional
+
+{microshift-short} now offers optional deployment of the built in logical volume manager storage(LVMS) Container Storage Interface(CSI) provider and CSI volume snapshot capabilities, reducing the use of runtime resources like RAM, CPU and storage.
+
+[id="lvms-stop-deployment_{context}"]
+== Stopping LVMS and CSI volume snapshot deployment
+
+Use the following procedure to prevent the deployment of LVMS CSI driver and CSI volume snapshot controller.
+
+.Procedure
+
+. To stop the LVMS CSI deployment, enter the following `driver` value under the `.storage` node in the {microshift-short} config file:
++
+[source,yaml]
+----
+# ...
+  storage
+   driver:'' <1>
+# ...
+----
+<1> Accepted values are `none` or `lvms`.
++
+[NOTE]
+====
+If the `driver` value is empty or null, {microshift-short} defaults to deploying LVMS.
+====
+
+. To stop CSI volume snapshot controller deployment, enter the following `optionalCsiComponents` value under the `.storage` node in the {microshift-short} config file:
++
+[source,yaml]
+----
+# ...
+  storage
+   optionalCsiComponents:'' <1>
+# ...
+----
+<1> Accepted values are `csi-snapshot-controller`, `csi-snapshot-webhook`, or `none`. `none` is mutually exclusive with all other values.
++
+[NOTE]
+====
+If the `optionalCsiComponents` value is empty or unspecified, {microshift-short} defaults to deploying snapshot-controller and snapshot-webhook.
+====
+
+. To restart {microshift-short} after the `.storage` node is specified with supported values in the {microshift-short} config file, run the following command:
++
+[source,terminal]
+----
+$ sudo systemctl restart microshift
+----
++
+[NOTE]
+====
+MicroShift does not redeploy the disabled components after restart.
+====
+
+[id="lvms-remove-installed-lvms_{context}"]
+== Removing installed LVMS and CSI volume snapshot
+
+To uninstall the deployed LVMS CSI driver and CSI volume snapshot controller, use the following procedure.
```

Comment:
I'm a bit confused here in that the last instructions I assumed a "deployment" ran on the CSI implementations, but here it reads like CSI implementations are deployments?

Reply:
@copejon need your suggestion

Reply:
I agree, "implementation" is more appropriate to the topic whereas "deployment" is already overloaded enough.

In the contributor docs I've been using "deployment" to generally mean "the installed CSI components." I ought to revisit them to make it more clear what we're referring to.

Reply:
change is done